app/jobs/course/material/zip_download_job.rb

```diff
@@ -29,6 +29,6 @@ def publish_file(file, name)
     public_file = File.join(public_dir, name)
     FileUtils.cp(file, public_file)
 
-    File.join(downloads_dir, job.id, File.basename(public_file))
+    URI.encode(File.join('/', downloads_dir, job.id, File.basename(public_file)))
```

Comment:
URI.join doesn't work well for this case, it expects the protocol and host. 

And string with white spaces will be considered as invalid URI too:

```
>   URI.join('/', 'foldername')
URI::BadURIError: both URI are relative

> URI.join('/', 'folder name')
URI::InvalidURIError: bad URI(is not URI?): folder name
```